crates/typst-syntax/src/package.rs

```diff
@@ -216,6 +216,171 @@ impl PackageVersion {
     }
 }
 
+impl PackageVersion {
+    // the code in this impl block is based on https://github.com/dtolnay/semver/blob/1.0.23/src/eval.rs
+    // author: David Tolnay <https://github.com/dtolnay>
+    // licensed under Apache 2.0
+
+    /// Performs an `==` match with the given version bound. Version elements missing in the bound
+    /// are ignored.
+    ///
+    /// ```
+    /// # use typst_syntax::package::{PackageVersion, SemVerBound};
+    /// # use std::str::FromStr;
+    /// let v1_1_1 = PackageVersion::from_str("1.1.1").unwrap();
+    ///
+    /// assert!(!v1_1_1.matches_eq(&SemVerBound::from_str("0").unwrap()));
+    /// assert!(v1_1_1.matches_eq(&SemVerBound::from_str("1").unwrap()));
+    /// assert!(!v1_1_1.matches_eq(&SemVerBound::from_str("2").unwrap()));
+    ///
+    /// assert!(!v1_1_1.matches_eq(&SemVerBound::from_str("1.0").unwrap()));
+    /// assert!(v1_1_1.matches_eq(&SemVerBound::from_str("1.1").unwrap()));
+    /// assert!(!v1_1_1.matches_eq(&SemVerBound::from_str("1.2").unwrap()));
+    ///
+    /// assert!(!v1_1_1.matches_eq(&SemVerBound::from_str("1.1.0").unwrap()));
+    /// assert!(v1_1_1.matches_eq(&SemVerBound::from_str("1.1.1").unwrap()));
+    /// assert!(!v1_1_1.matches_eq(&SemVerBound::from_str("1.1.2").unwrap()));
```

Comment:
The doc tests are a bit much imo. The total PR diff is a lot of lines for such a small thing. Also doc tests are slow to compile. I'd prefer a few (like ten assertions) in a `#[cfg(test)]` module at the bottom.